wallet/how-to/use-non-evm-networks/starknet/troubleshoot.md

```diff
@@ -0,0 +1,108 @@
+---
+description: Troubleshoot common Starknet issues.
+sidebar_position: 6
+---
+
+# Troubleshoot
+
+## 1. Connection issues
+
+### Wallet connection fails or doesn't respond.
+
+Ensure that MetaMask is installed and the StarkNet Snap is properly set up. You can check if the Snap is available using:
+
+```javascript
+const availableWallets = getAvailableWallets();
+if (!availableWallets.some(wallet => wallet.type === 'snap')) {
+  alert('Please install StarkNet Snap in MetaMask to proceed.');
+}
+```
+
+### The `connect()` function returns null or undefined.
+
+Verify that the Snap is installed and MetaMask has granted permission for the dapp to connect. Always include error handling in the connect method to manage issues:
+
+```javascript
+try {
+  const res = await connect();
+  if (!res) {
+    console.log('No wallet connected');
+  }
+} catch (error) {
+  console.error('Error connecting to wallet:', error);
+}
+```
+
+## 2. Snap permissions
+
+### Snap is not properly approved.
+
+Guide users through the process of approving the Snap in MetaMask. If they deny the connection, provide a retry option.
```

Comment:
Again this is handled automatically be get-starknet. When using the invokeSnaps what we describe in the previous section should already answer this. 

Specifically 


```typescript
provider
  .request({
    method: 'wallet_requestSnaps',
    params: {
      [snapId]: { version: snapVersion },
    },
  })
  .then(() => {
    // The snap has been installed proceed accordingly.
  })
  .catch(() => {
    // The snap has not been installed (user rejected the installation) 
    // Here we can provide retry options 
  });
```